x-pack/legacy/plugins/apm/public/components/app/ErrorGroupOverview/index.tsx

```diff
@@ -22,73 +23,124 @@ import { ErrorDistribution } from '../ErrorGroupDetails/Distribution';
 import { ErrorGroupList } from './List';
 import { useUrlParams } from '../../../hooks/useUrlParams';
 import { useTrackPageview } from '../../../../../infra/public';
+import { PROJECTION } from '../../../projections/typings';
+import { LocalUIFilters } from '../../shared/LocalUIFilters';
+import { useTransactionTypes } from '../../../hooks/useTransactionTypes';
 
 const ErrorGroupOverview: React.SFC = () => {
+  const { urlParams, uiFilters } = useUrlParams();
+
   const {
-    urlParams: { serviceName, start, end, sortField, sortDirection },
-    uiFilters
-  } = useUrlParams();
+    serviceName,
+    start,
+    end,
+    sortField,
+    sortDirection,
+    transactionType
+  } = urlParams;
 
   const { data: errorDistributionData } = useFetcher(() => {
     if (serviceName && start && end) {
       return loadErrorDistribution({
         serviceName,
+        transactionType,
         start,
         end,
         uiFilters
       });
     }
-  }, [serviceName, start, end, uiFilters]);
+  }, [serviceName, start, end, uiFilters, transactionType]);
 
   const { data: errorGroupListData } = useFetcher(() => {
     if (serviceName && start && end) {
       return loadErrorGroupList({
         serviceName,
+        transactionType,
         start,
         end,
         sortField,
         sortDirection,
         uiFilters
       });
     }
-  }, [serviceName, start, end, sortField, sortDirection, uiFilters]);
+  }, [
+    serviceName,
+    start,
+    end,
+    sortField,
+    sortDirection,
+    uiFilters,
+    transactionType
+  ]);
+
+  const transactionTypes = useTransactionTypes(urlParams);
 
-  useTrackPageview({ app: 'apm', path: 'error_group_overview' });
+  useTrackPageview({
+    app: 'apm',
+    path: 'error_group_overview'
+  });
   useTrackPageview({ app: 'apm', path: 'error_group_overview', delay: 15000 });
 
+  const localUIFiltersConfig = useMemo(
+    () => ({
+      filterNames: [
+        'transactionResult',
+        'host',
+        'containerId',
+        'podId'
+      ] as LocalUIFilterName[],
+      params: {
+        serviceName,
+        transactionType
+      },
+      projection: PROJECTION.ERROR_GROUPS
+    }),
+    [serviceName, transactionType]
+  );
+
   if (!errorDistributionData || !errorGroupListData) {
     return null;
   }
 
   return (
-    <React.Fragment>
-      <EuiFlexGroup>
-        <EuiFlexItem>
-          <EuiPanel>
-            <ErrorDistribution
-              distribution={errorDistributionData}
-              title={i18n.translate(
-                'xpack.apm.serviceDetails.metrics.errorOccurrencesChartTitle',
-                {
-                  defaultMessage: 'Error occurrences'
-                }
-              )}
-            />
-          </EuiPanel>
-        </EuiFlexItem>
-      </EuiFlexGroup>
-
-      <EuiSpacer size="s" />
+    <EuiFlexGroup>
+      <EuiFlexItem grow={1}>
+        <LocalUIFilters
+          {...localUIFiltersConfig}
```

Comment:
Agree that passing them as props directly reads better, but: we really need the memoization due to the `useFetcher` call (unless you can think of an alternative), and we can't use inline hooks because the render function exits early in some cases, and the linter will complain (maybe that's also why I've never seen people use it like that).